### PR TITLE
types: use flags in types package to handle clip zero case

### LINKED ISSUE #47518: use `FlagClipNegativeToZero` to handle clip to zero case in `types` package
one of the refactor issue of https://github.com/pingcap/tidb/issues/47355

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #47518

### What is changed and how it works?

This is one of the PR for code refactor, see: #47355

We use `FlagClipNegativeToZero` to indicate whether to clip the negative value to zero when casting the negative number to be unsigned.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessionctx/stmtctx/stmtctx.go [hunk 1]
```diff
@@ -1201,9 +1194,22 @@ func (sc *StatementContext) SetFlagsFromPBFlag(flags uint64) {
 	sc.TruncateAsWarning = (flags & model.FlagTruncateAsWarning) > 0
 	sc.InInsertStmt = (flags & model.FlagInInsertStmt) > 0
 	sc.InSelectStmt = (flags & model.FlagInSelectStmt) > 0
+	sc.InDeleteStmt = (flags & model.FlagInUpdateOrDeleteStmt) > 0
```
Comment (lcwangchao) [inline sessionctx/stmtctx/stmtctx.go:1197]: `SetFlagsFromPBFlag` looks the same with `flagsToStatementContext` in the mock store except it did not set `InDeleteStmt` (I think we missed it before). So I replace the implement in `flagsToStatementContext` by calling `SetFlagsFromPBFlag` directly

File: pkg/sessionctx/stmtctx/stmtctx.go [hunk 2]
```diff
@@ -1255,9 +1242,13 @@ func (sc *StatementContext) SetFlagsFromPBFlag(flags uint64) {
 	sc.TruncateAsWarning = (flags & model.FlagTruncateAsWarning) > 0
 	sc.InInsertStmt = (flags & model.FlagInInsertStmt) > 0
 	sc.InSelectStmt = (flags & model.FlagInSelectStmt) > 0
+	sc.InDeleteStmt = (flags & model.FlagInUpdateOrDeleteStmt) > 0
 	sc.OverflowAsWarning = (flags & model.FlagOverflowAsWarning) > 0
 	sc.IgnoreZeroInDate = (flags & model.FlagIgnoreZeroInDate) > 0
 	sc.DividedByZeroAsWarning = (flags & model.FlagDividedByZeroAsWarning) > 0
+	sc.SetTypeFlags(typectx.StrictFlags.
```
Comment (lcwangchao) [inline pkg/sessionctx/stmtctx/stmtctx.go:1249]: SetFlagsFromPBFlag looks the same with flagsToStatementContext in the mock store except it did not set InDeleteStmt (I think we missed it before). So I replace the implement in flagsToStatementContext by calling SetFlagsFromPBFlag directly

**Review by lance6716 — APPROVED** [review]

File: pkg/ddl/backfilling_scheduler.go [hunk 3]
```diff
@@ -148,12 +148,6 @@ func initSessCtx(
 	sqlMode mysql.SQLMode,
 	tzLocation *model.TimeZoneLocation,
 ) error {
-	// Unify the TimeZone settings in newContext.
-	if sessCtx.GetSessionVars().StmtCtx.TimeZone() == nil {
-		tz := *time.UTC
-		sessCtx.GetSessionVars().StmtCtx.SetTimeZone(&tz)
-	}
-	sessCtx.GetSessionVars().StmtCtx.IsDDLJobInQueue = true
```
Comment (YangKeao) [inline pkg/ddl/backfilling_scheduler.go:156]: Is this change expected? Though it seems that it's only used in `pkg/executor/ddl.go` 🤔 .

**Review by YangKeao — COMMENTED** [review]
Rest LGTM

File: pkg/ddl/backfilling_scheduler.go (hunk 3 above)
Comment (lcwangchao) [reply pkg/ddl/backfilling_scheduler.go:156]: I think `IsDDLJobInQueue` is used to make `ShouldClipToZeror` in the previous code. The verify CI passed after removing it.. @tangenta PTAL

Comment (YangKeao) [reply pkg/ddl/backfilling_scheduler.go:156]: IIUC, `pkg/executor/ddl.go` is not related with `pkg/ddl/backfilling_scheduler.go` 🤔 , so it seems fine for this case.

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**